src/sdk/strategy/pre-strategy.ts

```diff
@@ -65,26 +62,34 @@ export class PreStrategy {
   }
 
   public async deploy(
+    web3Provider: ethers.providers.Web3Provider,
     label: string,
-    provider: ethers.providers.Web3Provider
+    threshold = Math.floor(this.cohort.numUrsulas / 2) + 1,
+    shares = this.cohort.numUrsulas
   ): Promise<DeployedPreStrategy> {
-    const porterUri = this.cohort.configuration.porterUri;
-    const configuration = { porterUri };
-    const alice = Alice.fromSecretKey(
-      configuration,
-      this.aliceSecretKey,
-      provider
-    );
-    const bob = new Bob(configuration, this.bobSecretKey);
+    if (shares > this.cohort.numUrsulas) {
+      throw new Error(
+        `Threshold ${threshold} cannot be greater than the number of Ursulas in the cohort ${this.cohort.numUrsulas}`
+      );
+    }
+
+    const porterUri = this.cohort.porterUri;
+    const alice = Alice.fromSecretKey(this.aliceSecretKey);
+    const bob = new Bob(porterUri, this.bobSecretKey);
```

Comment:
Related to a previous comment - only Bob's public key should be needed for PRE, so secret key should not be needed.

Reply:
I think I'm missing something about this transition that we're going through, i.e. from `PreTDecDecrypter` to just `PreDecrypter`. For example, I thought that Bob needs a secret key to decrypt message kits at some point.